core/arch/arm/kernel/tee_i2c.c

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 Foundries Ltd <jorge@foundries.io>
+ *
+ * Brief   Access to linux i2c bus
+ *
+ */
+#include <kernel/tee_i2c.h>
+#include <kernel/thread.h>
+#include <mm/mobj.h>
+#include <optee_rpc_cmd.h>
+#include <string.h>
+
+#define I2C_BUFFER_LENGTH 512
+static struct mobj *mobj;
+
+TEE_Result tee_i2c_transfer(struct TEE_I2CRequest *req, size_t *bytes)
+{
+	TEE_Result res = TEE_SUCCESS;
+
+	if (!bytes || !req)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (req->bufferLen > I2C_BUFFER_LENGTH) {
+		EMSG("increase buffer size (%d)", (uint32_t)req->bufferLen);
+		return TEE_ERROR_BAD_PARAMETERS;
+	}
+
+	if (!mobj) {
+		mobj = thread_rpc_alloc_kernel_payload(I2C_BUFFER_LENGTH);
+		if (!mobj) {
+			EMSG("increase linux CONFIG_OPTEE_SHM_NUM_PRIV_PAGES");
```

Comment:
We are agnostic to the OS in normal world, why not just say that the kernel memory allocation failed?

Reply:
I thought it would help others...took me a few hours to find out how this config was affecting the driver. maybe I just add a comment in the code? would it be ok?